### ISSUE TITLE
Occasional Role Is 'null'

### ISSUE THREAD

**Issue by officialxian** [evanbrumley/spyfall#12]
When playing we periodically get someone whose role is 'null' which seems like it'd be the player gets assigned a role number outside of the array index, or that they never get a proper role assigned somehow in ./spyfall/client/main.js; function assignRoles; starting line 150. This is just speculation.

**Comment by galacticfan** [comment]
The shuffling code for the array looks correct and I can't see how it could assign a null unless there were more players than there are roles? i.e. there would need to be more than 8 players (max is 7 roles and 1 spy).
I know that I experience a null when I refresh the page, so where I leave and rejoin.

**Comment by leranus** [comment]
>  I know that I experience a null when I refresh the page, so where I leave and rejoin.
This happened to me also. We had to restart the game, so it would be great if this can be fixed.

**Comment by officialxian** [comment]
Not sure, but I don't think I've had more players than there are roles, I
checked that when I first saw it.

**Comment by evanbrumley** [comment]
Hey guys, thanks for the input. I believe this issue may have been caused by a race condition, as the code to distribute player roles is currently driven by the client rather than the server. Weird stuff can happen when multiple players tap 'start game' at the same time, and the problems have been compounded by some very heavy traffic lately. I've created ticket #25 to move the role generation code on to the server side, which will hopefully deal with the problem once and for all :-)
